lib/matplotlib/tests/test_figure.py

```diff
@@ -1010,18 +1010,27 @@ def test_axes_kwargs():
     plt.close()
 
     # plt.subplot() searches for axes with the same subplot spec, and if one
-    # exists, returns it, regardless of whether the axes kwargs were the same.
+    # exists, and the kwargs match returns it, create a new one if they do not
     fig = plt.figure()
     ax = plt.subplot(1, 2, 1)
     ax1 = plt.subplot(1, 2, 1)
     ax2 = plt.subplot(1, 2, 2)
+    # This will delete ax / ax1 as they fully overlap
```

Comment:
Isn't the reason for deletion actually that the projection keywords are not the same?

Reply:
That is what I thought too, but the logic is https://github.com/matplotlib/matplotlib/blob/ea6803254567eaaa9adb027b541a6885d9fa246f/lib/matplotlib/pyplot.py#L1236-L1244 which says "delete any axes which the now axes fully covers" .

Reply:
@tacaswell is correct. I wasn't aware of this behavior, but it's well documented, and it's been around a long time, so this is 👍 for me.